crates/matrix-sdk/src/notification_settings/rule_commands.rs

```diff
@@ -482,4 +502,49 @@ mod tests {
             );
         }
     }
+
+    #[async_test]
+    async fn test_set_rule_actions() {
+        let mut ruleset = get_server_default_ruleset();
+        let mut rule_commands = RuleCommands::new(ruleset.clone());
+
+        // Starting with an empty action list for `PredefinedUnderrideRuleId::Message`.
+        ruleset
+            .set_actions(RuleKind::Underride, PredefinedUnderrideRuleId::Message, vec![])
+            .unwrap();
+
+        // After setting a list of actions
+        rule_commands
+            .set_rule_actions(
+                RuleKind::Underride,
+                PredefinedUnderrideRuleId::Message.as_str(),
+                vec![Action::Notify, Action::SetTweak(Tweak::Sound("default".into()))],
+            )
+            .unwrap();
+
+        // The ruleset must have been updated
+        let actions = rule_commands
+            .rules
+            .get(RuleKind::Underride, PredefinedUnderrideRuleId::Message)
+            .unwrap()
+            .actions();
+        assert_eq!(actions.len(), 2);
+
+        // and a `SetPushRuleActions` command must have been added
+        assert_eq!(rule_commands.commands.len(), 1);
+        assert_matches!(&rule_commands.commands[0],
+            Command::SetPushRuleActions { scope, kind, rule_id, actions } => {
+                assert_eq!(scope, &RuleScope::Global);
+                assert_eq!(kind, &RuleKind::Underride);
+                assert_eq!(rule_id, PredefinedUnderrideRuleId::Message.as_str());
+                assert_eq!(actions.len(), 2);
+                assert_matches!(&actions[0], Action::Notify);
+                assert_matches!(&actions[1], Action::SetTweak(tweak) => {
+                    assert_matches!(tweak, Tweak::Sound(sound) => {
+                        assert_eq!(sound, "default");
+                    });
+                });
```

Comment:
nit: it's possible to at least flatten by one level, if not two:

```suggestion
                assert_matches!(&actions[1], Action::SetTweak(Tweak::Sound(sound))) => {
                        assert_eq!(sound, "default");
                });
```

if not:


```suggestion
                assert_matches!(&actions[1], Action::SetTweak(Tweak::Sound("default"))) => {});
```

Reply:
This one last form won't work. The first form is the one that must be used here.

Reply:
👍